sql/core/src/main/scala/org/apache/spark/sql/catalog/interface.scala

```diff
@@ -64,12 +64,26 @@ class Database(
 @Stable
 class Table(
     val name: String,
-    @Nullable val database: String,
+    @Nullable val qualifier: Array[String],
```

Comment:
After more thought, Maybe 2 parameters is better here: `catalog: String, namespace: Array[String]`. The point is, for temp views, they don't belong to any catalog, and setting `catalog` to null is more clear.